### PR TITLE
Notifications serialise connect, disconnect, block_added

### PR DESCRIPTION
**Pull request by ShahanaFarooqui**
- Serialise `connect`, `discnnect` and `block_added`.

Changelog-Added: Plugins: connect, disconnect, block_added notifications now have fields in sub-object of same name (e.g. "connect": {...}), making all notifications uniform
Changelog-Deprecated: Plugins: connect, disconnect, block_added notifications fields outside the sub-object of the same name: e.g. parse .connect.id instead of .id.

### CODE REVIEW

File: lightningd/notification.c [hunk 1]
```diff
@@ -575,7 +599,7 @@ void notify_balance_snapshot(struct lightningd *ld,
 static void block_added_notification_serialize(struct json_stream *stream,
 					       struct block *block)
 {
-	json_object_start(stream, "block");
+	json_object_start(stream, "block_added");
```
Comment (rustyrussell) [inline lightningd/notification.c:602]: I see your motivation here, but I think you need to do *both* for compatibility?

File: doc/guides/Developer-s Guide/plugin-development/event-notifications.md [hunk 2]
```diff
@@ -378,7 +382,7 @@ Emitted after we've caught up to the chain head on first start. Lists all curren
 
 ```json
 {
-    "balance_snapshots": [
+    "balance_snapshot": [
```
Comment (rustyrussell) [inline doc/guides/Developer-s Guide/plugin-development/event-notifications.md:385]: I can't see that you actually made this change?

File: doc/guides/Developer-s Guide/plugin-development/event-notifications.md [hunk 3]
```diff
@@ -87,9 +87,11 @@ A notification for topic `connect` is sent every time a new connection to a peer
 
 ```json
 {
```
Comment (rustyrussell) [inline doc/guides/Developer-s Guide/plugin-development/event-notifications.md:89]: We should talk about the form of all these notifications (i.e. that notification X will contain an object X) in the general section of this doc.

Should we remove PLUGINS.md too, as a previous commit?  It's confusing to have both!

**Review by rustyrussell — CHANGES_REQUESTED** [review]

File: lightningd/notification.c (hunk 1 above)
Comment (ShahanaFarooqui) [reply lightningd/notification.c]: Sorry, forgot to add deprecated_apis logic. Updated the code to keep it compatible with commit 769e59f472402b1892699bc981acc382aa198a8d

File: doc/guides/Developer-s Guide/plugin-development/event-notifications.md (hunk 2 above)
Comment (ShahanaFarooqui) [reply doc/guides/Developer-s Guide/plugin-development/event-notifications.md:385]: It is the typo in the documentation only. Event name and response both are `balance_snapshot` but document shows it as `balance_snapshots`

File: doc/guides/Developer-s Guide/plugin-development/event-notifications.md (hunk 3 above)
Comment (ShahanaFarooqui) [reply doc/guides/Developer-s Guide/plugin-development/event-notifications.md:89]: Agree, it is redundant but we can leave it till I am not working on documentation consolidation. I will work on the logical placement of all these documents at that point anyways.

**Review by rustyrussell — COMMENTED** [review]
Changelog-Added: Plugins: `connect`, `disconnect`, `block_added` notifications now have fields in sub-object of same name (e.g. `"connect": {...}`), making all notifications uniform
Changelog-Deprecated: Plugins: `connect`, `disconnect`, `block_added` notifications fields outside the sub-object of the same name: e.g. parse .connect.id instead of .id.

**Pull request closed without merging**